swaybar/main.c

```diff
@@ -362,10 +401,32 @@ void render() {
 	// Command output
 	cairo_set_source_u32(window->cairo, colors.statusline);
 	int width, height;
-	get_text_size(window, &width, &height, "%s", line);
 
-	cairo_move_to(window->cairo, window->width - margin - width, margin);
-	pango_printf(window, "%s", line);
+	if (protocol == TEXT) {
+		get_text_size(window, &width, &height, "%s", line);
+		cairo_move_to(window->cairo, window->width - margin - width, margin);
+		pango_printf(window, "%s", line);
+	} else if (protocol == I3BAR && status_line) {
+		int i;
+		int moved = 0;
+		bool corner = true;
+		for ( i = status_line->length - 1; i >= 0; --i ) {
+			struct status_block *block = status_line->items[i];
+			if (block->full_text && block->full_text[0]) {
+				get_text_size(window, &width, &height, "%s", block->full_text);
+				moved += width + block->separator_block_width;
+				cairo_move_to(window->cairo, window->width - margin - moved, margin);
+				cairo_set_source_u32(window->cairo, block->color);
+				pango_printf(window, "%s", block->full_text);
+				if (corner) {
+					corner = false;
+				} else if (block->separator) {
+					cairo_move_to(window->cairo, window->width - margin - moved + width, margin);
+					pango_printf(window, "|");
```

Comment:
My i3 prints a grey vertical line.
Printing a pipe is not exactly the same, but it looks similar enough. I probably should change the color and use a uniform grey? instead of the field color.